src/test/resources/examples/discriminatedOneOf/api.yaml

```diff
@@ -0,0 +1,40 @@
+openapi: 3.0.0
+info:
+paths:
+components:
+  schemas:
+    SomeObj:
+      type: object
+      required:
+        - state
+      properties:
+        state:
+          $ref: '#/components/schemas/State'
+    State:
+      oneOf:
+        - $ref: '#/components/schemas/StateA'
+        - $ref: '#/components/schemas/StateB'
+      discriminator:
+        propertyName: status
+        mapping:
```

Comment:
I think we need to expand this api spec to include some more examples. If you add a second case where the discriminator has no `mapping` object, it seems to fall over

Reply:
I added this case and also fixed it in the last commit